src/test/java/uk/nhs/digital/nhsconnect/nhais/translator/AmendmentPreviousNameToEdifactMapperTest.java

```diff
@@ -0,0 +1,88 @@
+package uk.nhs.digital.nhsconnect.nhais.translator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.FhirValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.PersonPreviousName;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBody;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentValue;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+import uk.nhs.digital.nhsconnect.nhais.translator.amendment.AmendmentPreviousNameToEdifactMapper;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class, SoftAssertionsExtension.class})
+class AmendmentPreviousNameToEdifactMapperTest extends AmendmentFhirToEdifactTestBase {
+
+    private static final String PREVIOUS_SURNAME = "Snow";
+
+    private final AmendmentPreviousNameToEdifactMapper translator = new AmendmentPreviousNameToEdifactMapper();
+
+    @Mock
+    private AmendmentBody amendmentBody;
+
+    @Mock
+    private JsonPatches jsonPatches;
+
+    @BeforeEach
+    void setUp() {
+        reset(amendmentBody, jsonPatches);
+
+        when(amendmentBody.getJsonPatches()).thenReturn(jsonPatches);
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "getAddOrReplaceEnums")
+    void whenAddingOrReplacingPreviousSurname_expectAllFieldsAreMapped(AmendmentPatchOperation operation) {
+        when(jsonPatches.getPreviousSurname()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation).setValue(AmendmentValue.from(PREVIOUS_SURNAME))));
+
+        var segments = translator.map(amendmentBody);
+
+        assertThat(segments).isPresent().get()
+            .isEqualTo(PersonPreviousName.builder()
+                .previousFamilyName(PREVIOUS_SURNAME)
+                .build());
+    }
```

Comment:
(Optional) Is Mockito really needed here? Wouldn't creating normal objects be easier to read and maintain?:
````java
List<AmendmentPatch> patches = List.of(
                new AmendmentPatch()
                    .setOp(operation)
                    .setValue(AmendmentValue.from(PREVIOUS_SURNAME))
				);
JsonPatches jsonPatches = new JsonPatches(amendmentBody, patches);//it's not good that we put amendmentBody there
when(amendmentBody.getJsonPatches()).thenReturn(jsonPatches)

var segments = translator.map(amendmentBody);

assertThat(segments).isPresent().get()
	.isEqualTo(PersonPreviousName.builder()
                .previousFamilyName(PREVIOUS_SURNAME)
                .build());
````

Reply:
I prefer leaving it as mocks. If someone changes implementation of those objects we might end up with a lot of refactor just to conform some new constructor param that isn't needed. There's been plenty of times when I started using normal objects and ended up with creating a really big tree of nested objects just to use 1 property.